torch/utils/collect_env.py

```diff
@@ -225,6 +234,131 @@ def get_nvidia_smi():
+                if pkg_name != "":
+                    cmd = f"{mgr_name} -l | grep {pkg_name}"
+            if cmd != "":
+                ret = run_and_read_all(run_lambda, cmd)
+    lst = []
+    if ret:
+        lst += re.sub(" +", " ", ret).split(" ")
+    if len(lst) > index and index != -1:
+        ret = lst[index]
+    else:
+        ret = "N/A"
+    return ret
+
+
+def get_intel_gpu_driver_version(run_lambda):
+    lst = []
+    platform = get_platform()
+    if platform == "linux":
+        for pkg in ["intel_opencl", "level_zero"]:
+            lst.append(f"* {pkg}:\t{get_pkg_version(run_lambda, pkg)}")
+    if platform == "win32" or platform == "cygwin":
+        txt = run_and_read_all(
+            run_lambda,
+            'powershell.exe "gwmi -Class Win32_PnpSignedDriver | where{$_.DeviceClass -eq \\"DISPLAY\\"\
+            -and $_.Manufacturer -match \\"Intel\\"} | Select-Object -Property DeviceName,DriverVersion,DriverDate\
+            | ConvertTo-Json"',
+        )
+        try:
+            obj = json.loads(txt)
+            if type(obj) is list:
+                for o in obj:
+                    lst.append(
+                        f'* {o["DeviceName"]}: {o["DriverVersion"]} ({o["DriverDate"]})'
+                    )
+            else:
+                lst.append(f'* {obj["DriverVersion"]} ({obj["DriverDate"]})')
+        except ValueError as e:
+            lst.append(txt)
+            lst.append(str(e))
+    return "\n".join(lst)
+
+
+def get_intel_gpu_onboard(run_lambda):
+    lst = []
+    platform = get_platform()
+    if platform == "linux":
+        txt = run_and_read_all(run_lambda, "xpu-smi discovery -j")
+        if txt:
+            try:
+                obj = json.loads(txt)
+                if type(obj["device_list"]) is list:
+                    for o in obj["device_list"]:
+                        lst.append(f'* {o["device_name"]}')
+                else:
+                    lst.append("N/A")
+            except (ValueError, TypeError) as e:
+                lst.append(str(e))
+        else:
+            lst.append("N/A")
+    if platform == "win32" or platform == "cygwin":
+        txt = run_and_read_all(
+            run_lambda,
+            'powershell.exe "gwmi -Class Win32_PnpSignedDriver | where{$_.DeviceClass -eq \\"DISPLAY\\"\
+            -and $_.Manufacturer -match \\"Intel\\"} | Select-Object -Property DeviceName | ConvertTo-Json"',
+        )
+        try:
+            obj = json.loads(txt)
+            if type(obj) is list:
+                for o in obj:
+                    lst.append(f'* {o["DeviceName"]}')
+            else:
+                lst.append(f'* {obj["DeviceName"]}')
+        except ValueError as e:
+            lst.append(txt)
+            lst.append(str(e))
+    return "\n".join(lst)
+
+
+def get_intel_gpu_detected(run_lambda):
+    if TORCH_AVAILABLE:
+        devices = [
+            f"* [{i}] {torch.xpu.get_device_properties(i)}"
```

Comment:
I think it is enough for us to know the device name rather than all device properties.

Reply:
device properties contains device name

Reply:
I mean `torch.xpu.get_device_properties()` seems a bit verbose.